src/Tasks/Microsoft.NET.Build.Tasks/ArrayBufferWriter.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    /// <summary>
+    /// Implementation of System.Buffers.IBufferWriter backed by ArrayPool.Shared.
+    /// </summary>
+    internal class ArrayBufferWriter : IBufferWriter<byte>, IDisposable
```

Comment:
please add a comment like `// TODO Remove once we have https://github.com/dotnet/corefx/issues/34894`

Reply:
Honestly, spreading this to every repo is getting silly. Let's wait for it to be in. I don't understand the rush to use this API. Particularly because we can't actually remove N.J until NuGet does.